src/rendergraph/common/rendergraph/texture.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <QImage>
+#include <memory>
+
+#include "backend/texture.h"
+
+namespace rendergraph {
+class Context;
+class Texture;
```

Comment:
as a reader, I now question whether you meant to implement `Context` in this class and forgot or if its just a necessary forward declaration (as opposed to the unnecessary forward declaration of `Texture`).

Reply:
I will include context.h for clarity.

Reply:
done. and at other locations where forward declaration are done out of necessity (to avoid circular dependency), I added a comment. Please resolve if okay.